repos/spack_repo/builtin/packages/py_charset_normalizer/package.py

```diff
@@ -24,13 +25,17 @@ class PyCharsetNormalizer(PythonPackage):
     version("2.0.7", sha256="e019de665e2bcf9c2b64e2e5aa025fa991da8720daa3c1138cadd2fd1856aed0")
 
     with default_args(type="build"):
+        depends_on("py-setuptools@68:81", when="@3.4.4:")
         depends_on("py-setuptools")
-        depends_on("py-setuptools-scm", when="@3.4.2:")
-        # Optional dependency when building with --use-mypyc
-        # depends_on("py-mypy@1.4.1:1.15.0", when="@3.4.2:")
+
+        # Historical dependencies
+        depends_on("py-setuptools-scm", when="@3.4.1:3.4.3")
+        depends_on("py-mypy@1.4.1:1.15.0", when="@3.4.2")
+        depends_on("py-mypy@1.4.1:1.14.0", when="@3.4.1")
 
     with default_args(type=("build", "run")):
-        depends_on("python@3.7:3.13", when="@3.4:")
+        depends_on("python@3.8:3.14", when="@3.4.3:")
```

Comment:
```suggestion
        depends_on("python@3.7:3.14", when="@3.4.3:")
```
according to [`pyproject.toml#L38`](https://github.com/jawah/charset_normalizer/blob/3.4.4/pyproject.toml#L38)

Reply:
3.8 according to https://pypi.org/project/charset-normalizer/#files. At the very least, 3.7 is no longer being tested or supported. Neither version is supported in Spack. I could also remove the lower bounds if you want to make things simpler.

Reply:
Thanks for the clarification. Can you then add a comment that the version restriction comes from the wheels listed in PyPI to prevent confusion in the future?